vcd/resource_vcd_nsxt_edgegateway_bgp_configuration_test.go

```diff
@@ -0,0 +1,628 @@
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "enabled", "false"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "local_as_number", "65420"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "graceful_restart_mode", "GRACEFUL_AND_HELPER"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "ecmp_enabled", "false"),
+				),
+			},
+		},
+	})
+	postTestChecks(t)
+}
+
+const testAccVcdNsxtBgpConfigPrereqs = `
+data "vcd_org_vdc" "nsxt-vdc" {
+  org  = "{{.Org}}"
+  name = "{{.NsxtVdc}}"
+}
+
+data "vcd_nsxt_edgegateway" "testing" {
+  org      = "{{.Org}}"
+  owner_id = data.vcd_org_vdc.nsxt-vdc.id
+
+  name = "{{.NsxtEdgeGw}}"
+}
+`
+
+const testAccVcdNsxtBgpConfig1 = testAccVcdNsxtBgpConfigPrereqs + `
+resource "vcd_nsxt_edgegateway_bgp_configuration" "testing" {
+  org = "{{.Org}}"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.testing.id
+
+  enabled                = false
+  local_as_number        = "65420"
+  graceful_restart_mode  = "GRACEFUL_AND_HELPER"
+  graceful_restart_timer = 190
+  stale_route_timer      = 500
+  ecmp_enabled           = false
+}
+`
+
+const testAccVcdNsxtBgpConfig2 = testAccVcdNsxtBgpConfigPrereqs + `
+resource "vcd_nsxt_edgegateway_bgp_configuration" "testing" {
+  org = "{{.Org}}"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.testing.id
+
+  enabled                = true
+  local_as_number        = "0.65420"
+  graceful_restart_mode  = "DISABLE"
+  graceful_restart_timer = 190
+  stale_route_timer      = 600
+  ecmp_enabled           = true
+}
+`
+
+const testAccVcdNsxtBgpConfig3DS = testAccVcdNsxtBgpConfig2 + `
+# skip-binary-test: Data Source test
+data "vcd_nsxt_edgegateway_bgp_configuration" "testing" {
+  org = "{{.Org}}"
+  
+  edge_gateway_id = data.vcd_nsxt_edgegateway.testing.id
+}
+`
+
+const testAccVcdNsxtBgpConfig4 = testAccVcdNsxtBgpConfigPrereqs + `
+resource "vcd_nsxt_edgegateway_bgp_configuration" "testing" {
+  org = "{{.Org}}"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.testing.id
+
+  enabled                = false
+  local_as_number        = "65430"
+  graceful_restart_mode  = "HELPER_ONLY"
+  graceful_restart_timer = 190
+  stale_route_timer      = 600
+  ecmp_enabled           = true
+}
+`
+
+const testAccVcdNsxtBgpConfig5 = testAccVcdNsxtBgpConfigPrereqs + `
+resource "vcd_nsxt_edgegateway_bgp_configuration" "testing" {
+  org = "{{.Org}}"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.testing.id
+
+  enabled                = true
+  local_as_number        = "65420"
+  graceful_restart_mode  = "GRACEFUL_AND_HELPER"
+  graceful_restart_timer = 190
+  stale_route_timer      = 600
+  ecmp_enabled           = true
+}
+`
+
+const testAccVcdNsxtBgpConfig6DS = testAccVcdNsxtBgpConfig5 + `
+# skip-binary-test: Data Source test
```

Comment:
Is this comments still valid?

Reply:
Yes. We can't create a resource and a data source referencing it in a single config file. For that reason, there are multiple configuration definitions (ending with `DS`) which will be skipped in binary tests, but will run in acceptance tests.